File: p_028_1_intentextras/src/main/java/com/startandroid/p_028_1_intentextras/MainActivity.java
package com.startandroid.p_028_1_intentextras;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    EditText et1, et2;
    Button btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);

        et1 = findViewById(R.id.et_1);
        et2 = findViewById(R.id.et_2);
        btn = findViewById(R.id.btn);

        btn.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        Intent intent = new Intent(this, ViewActivity.class);
        intent.putExtra("firstname", et1.getText().toString());
        intent.putExtra("lastname", et2.getText().toString());
        startActivity(intent);

    }
}

File: k_093_service_stop/src/main/java/com/startandroid/k_093_service/MainActivity.kt
package com.startandroid.k_093_service

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.start_btn).setOnClickListener {
            startService(Intent(this, MyService::class.java).putExtra("time",7))
            startService(Intent(this, MyService::class.java).putExtra("time",2))
            startService(Intent(this, MyService::class.java).putExtra("time",4))
        }
    }
}